subsys/net/lib/openthread/platform/uart.c

```diff
@@ -217,3 +226,35 @@ otError otPlatUartSend(const u8_t *aBuf, u16_t aBufLength)
 		return OT_ERROR_BUSY;
 	}
 };
+
+otError otPlatUartFlush(void)
+{
+	u32_t len;
+	const u8_t *data;
+	otError result = OT_ERROR_NONE;
+
+	len = ring_buf_get_claim(ot_uart.tx_ringbuf, (u8_t **)&data,
```

Comment:
Not exactly my area of expertise, but shouldn't we call `ring_buf_get_finish` after we process the data? In the API docs we have:
> Once data is processed it can be freed using @ref ring_buf_get_finish.

Also, shouldn't we call this in a loop until `ring_buf_is_empty()` returns true in order to fully flush the buffer? IIUC `ring_buf_get_claim` returns the largest continuous buffer it can, but it isn't necessarily all of the data available in the buffer (in case the data wraps). At least, that's what documentation suggests (and brief look into the code confirms that):
>   @return Number of valid bytes in the provided buffer which can be smaller than requested if there is not enough free space or **buffer wraps**.

Reply:
According to documentation, you are right about the need to free space and call claim operation in loop. So I implemented do while() loop, which should solve the problem, please verify it. If it comes to usage the ring_buf_is_empty() method, it leads to lock in infinite loop, as buffer is not empty even when ring_buf_get_claim() can't get anything from it and returns 0 bytes every time. I'm not sure why is it happening, so I look to the ring buffer tests, where buffer empty is checked by trying to claim something and getting 0 bytes in result and ring_buf_is_empty() is not used for that. I thought about leaving it implemented this way, as it is compatible with tests, but also to contact with verification team to ask about this problem. Is this ok?

Reply:
If it is so, then indeed it seems something fishy is going on with `ring_buf_is_empty()` function, probably worth an Zephyr issue to create. Nonetheless, current solution looks fine to me.